test/e2e_env/kubernetes/meshtrafficpermission/api.go

```diff
@@ -0,0 +1,75 @@
+package meshtrafficpermission
```

Comment:
Do we really need an e2e test for creating every resource? Maybe it's just the state of the framework but this test feels so noisy. What is this testing/guarding against exactly?

Reply:
This one is just for me to feel confident policy-gen works as expected because it's not really tested anywhere else. I promise not to create similar tests for other policies :D

Reply:
Hmm ok, might be worth a comment